### ISSUE TITLE
[Improvement] Panic channel should use sref (service reference) instead of channel number

### ISSUE THREAD

**Issue by wedebe** [openatv/enigma2#2701]
Using channel number for panic zap can lead to an unintended channel being zapped to, if the bouquet has had channels added or removed since the user set the original value.

**Comment by geeeeeeeeeee** [comment]
Using service reference for panic zap, can't zap, if the service reference doesn't exist anymore...

?? config.servicelist.startupservice != config.usage.panicchannel

**Comment by wedebe** [comment]
Perhaps notifying the user when they're about to remove a panic channel (same should be done for startup channel) would be a solution? Or, more simply, storing the sref in prefs and falling back to the channel number?

```
config.usage.panicbutton=True
config.usage.panicchannel=119
config.usage.panicsref=1:0:1:178F:7D7:2:11A0000:0:0:0
```

**Comment by kiddac** [comment]
I agree with above. Surplus unnecessary junk. Why not just totally remove it. One less option in the mountain of settings.

**Comment by Schimmelreiter** [comment]
WTF is a panic channel?

**Comment by jbleyel** [comment]
This feature request has a real purpose, because the channel number is not always the same channel.

**Comment by jbleyel** [comment]
> WTF is a panic channel?

It's similar like the boss-key.

**Comment by geeeeeeeeeee** [comment]
But that is for purpuse. Than Channel 1 ist better than a wrong ref.

**Comment by wedebe** [comment]
> And you really use it?

Yes I do



> I agree with above. Surplus unnecessary junk. Why not just totally remove it. One less option in the mountain of settings.

Just because you don't use it doesn't mean that other people don't. Your advice almost amounts to "why do you use Linux equipment, just get a Sky Box" 😆



> WTF is a panic channel?

Pressing `0` will jump to a determined channel in a specific bouquet. It's not just for a _panic_ situation (eg. watching porn when someone suddenly comes 
into the room); it can also save multiple button presses of opening the channel list, changing bouquet, then changing to your most-watched channel. 

> It works and there is no change needed! Bad request only!

Can you explain how it works when it doesn't?


I've since had a further thought on this... how is the `start-up channel` handled? Does it also happen there? I haven't had time to look into the code or test this.

**Comment by geeeeeeeeeee** [comment]
Sorry, but you don't like to change your most-watched channel from 119 to 1?

**Comment by jbleyel** [comment]
Just for clarification:

These are the descriptions for the panic settings:


```
"Enable panic button" / "When enabled, pressing '0' will zap you to the first channel in your first bouquet and delete your zap-history."
"Panic channel" / "Pressing the panic button will zap you to this channel number in your first bouquet."

```

**Comment by geeeeeeeeeee** [comment]
I had an idea but it's maybe like a hack.
Set the number to `0000` could use the startupservice with sref.
